### PR TITLE
fix/11427: new design for change password when user is not validated

### PR DESCRIPTION
**Pull request by tienifr**
### Details

When user didn't validate account, we should show resendValidationLink screen

### Fixed Issues

$ https://github.com/Expensify/App/issues/11427

### Tests

1. Create account in Old dot
2. Click on Free plan inbox card
3. Stop loading redirected page
4. Change host `https://new.expensify.com` to `http://localhost:8080/`
5. Navigate to Settings > Security > change password
6. Check `UnauthenticatedUser` screen and click button `Resend validation link`
7. Check the error message if Error exist, otherwise success message

- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. Create account in Old dot
2. Click on Free plan inbox card
3. Navigate to Settings > Security > change password
4. Check `UnauthenticatedUser` screen and click button `Resend validation link`
5. Check the error message if Error exist, otherwise success message

- [x] Verify that no errors appear in the JS console

#### Web

Screen Shot 2022-10-20 at 10 34 51
Screen Shot 2022-10-20 at 10 35 15

#### Mobile Web - Chrome

IMG_4824
IMG_4825

#### Mobile Web - Safari

IMG_4822

IMG_4823

### CODE REVIEW

**Comment by tienifr** [comment]
@rushatgabhane Sorry about that. I'll keep that in my mind. Thanks

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

File: src/pages/signin/ResendValidationForm.js [hunk 1]
```diff
@@ -78,21 +88,24 @@ const ResendValidationForm = (props) => {
...
-                <Button
-                    medium
-                    success
-                    text={props.translate('resendValidationForm.resendLink')}
-                    isLoading={props.account.isLoading}
-                    onPress={() => (props.account.validated ? Session.resendResetPassword() : Session.resendValidationLink())}
-                    isDisabled={props.network.isOffline}
-                />
-            </View>
+            {!props.shouldHideButton
```
[the remark this answers is not in the document]
Comment (tienifr) [reply src/pages/signin/ResendValidationForm.js:91]: @rushatgabhane Thanks for your review. In `EditPasswordScreen` we just validate data when users click save button. But if we use `Form` component, we'll validate whenever user change input. That makes action `shouldShowNewPasswordPrompt` and `ErrorUtils.getLatestErrorMessage(this.props.account)` have different behavior. Pls let me know what you think.

File: src/pages/settings/EditPasswordScreen.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                    </Text>
+                    <View style={styles.mb6}>
+                        <TextInput
+                            label={`${this.props.translate('passwordPage.currentPassword')}*`}
+                            ref={el => this.currentPasswordInputRef = el}
+                            secureTextEntry
+                            autoCompleteType="password"
+                            textContentType="password"
+                            value={this.state.currentPassword}
+                            onChangeText={text => this.clearErrorAndSetValue('currentPassword', text)}
```
[the remark this answers is not in the document]
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:156]: @rushatgabhane Thanks for your review. In `EditPasswordScreen` we just validate data when users click save button. But if we use `Form` component, we'll validate whenever user change input. That makes action `shouldShowNewPasswordPrompt` and `ErrorUtils.getLatestErrorMessage(this.props.account)` have different behavior. Pls let me know what you think.

[the remark this answers is not in the document]
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:156]: @rushatgabhane 
- Current showing password prompt logic:

https://user-images.githubusercontent.com/113963320/198529168-df7d782d-42d8-4719-805a-31e29e4bc5cb.mov

we just detect the error and hide passwordPrompt when user clicks save button

- Result if I change to use Form component:

https://user-images.githubusercontent.com/113963320/198529483-cee29150-0d36-4aa6-970b-1c09cbd70cda.mov

The error will be detected and hide passwordPrompt when newPassword failed

[the remark this answers is not in the document]
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:156]: @rushatgabhane we'll remove logic `_.every(this.state.errors, error => !error)` in https://github.com/Expensify/App/blob/main/src/pages/settings/PasswordPage.js#L211 as well, right?

[the remark this answers is not in the document]
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:156]: Ok, It seems to be good. I'll integrate Form component, Thanks

**Comment by tienifr** [comment]
@rushatgabhane I've updated my PR. I changed to use Form component, and updated texts as https://github.com/Expensify/App/issues/11427#issuecomment-1293843676. Pls help check. Thanks

**Comment by luacmartins** [comment]
@rushatgabhane let me know once you are done testing!

File: src/libs/Navigation/AppNavigator/ModalStackNavigators.js [hunk 3]
```diff
@@ -231,6 +231,13 @@ const SettingsModalStackNavigator = createModalStackNavigator([
         },
         name: 'Settings_Password',
     },
+    {
+        getComponent: () => {
+            const SettingsPasswordPage = require('../../../pages/settings/ResendValidationLinkScreen').default;
```
Comment (luacmartins) [inline src/libs/Navigation/AppNavigator/ModalStackNavigators.js:236]: This component is in `pages/settings` so let's change the name to Page instead of Screen
```suggestion
            const SettingsPasswordPage = require('../../../pages/settings/ResendValidationLinkPage').default;
```

File: src/libs/actions/Session/index.js [hunk 4]
```diff
@@ -565,6 +565,11 @@ function authenticatePusher(socketID, channelName, callback) {
     });
 }
 
+function updateSessionLoginAndResendValidationLink(login) {
+    Onyx.merge(ONYXKEYS.CREDENTIALS, {login});
```
Comment (luacmartins) [inline src/libs/actions/Session/index.js:569]: We should move this to the optimistData in resendValidationLink - https://github.com/Expensify/App/blob/44a6876023facd8c0620424c9538aef676e6ee05/src/libs/actions/Session/index.js#L91-L99

File: src/pages/settings/EditPasswordScreen.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
+import PropTypes from 'prop-types';
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:1]: Let's rename this file to `EditPasswordPage`

File: src/pages/settings/EditPasswordScreen.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    account: PropTypes.shape({
+        /** An error message to display to the user */
+        errors: PropTypes.objectOf(PropTypes.string),
+
+        /** Success message to display when necessary */
+        success: PropTypes.string,
+
+        /** Whether a sign on form is loading (being submitted) */
+        isLoading: PropTypes.bool,
+    }),
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:28]: This data now comes from the EDIT_PASSWORD_FORM key, no? DO we still need to subscribe to account?

File: src/pages/settings/EditPasswordScreen.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    account: {},
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+class EditPasswordScreen extends Component {
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:37]:
```suggestion
class EditPasswordPage extends Component {
```

File: src/pages/settings/EditPasswordScreen.js [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    constructor(props) {
+        super(props);
+        this.submit = this.submit.bind(this);
+        this.validate = this.validate.bind(this);
+
+        this.errorKeysMap = {
+            currentPassword: 'passwordPage.errors.currentPassword',
+            newPasswordSameAsOld: 'passwordPage.errors.newPasswordSameAsOld',
+            newPassword: 'passwordPage.errors.newPassword',
+        };
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:47]: Why use this map vs just inline paths?

File: src/pages/settings/EditPasswordScreen.js [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    render() {
+        return (
+            <ScrollView
+                style={styles.flex1}
+                contentContainerStyle={styles.p5}
+
+                // Allow the user to click show password while password input is focused.
+                // eslint-disable-next-line react/jsx-props-no-multi-spaces
+                keyboardShouldPersistTaps="always"
+            >
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:86]: This will probably cause some issues since Form already wraps its children in a ScrollView.

File: src/pages/settings/EditPasswordScreen.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+                <Text style={[styles.mb6]}>
+                    {this.props.translate('passwordPage.changingYourPasswordPrompt')}
+                </Text>
+                <Form
+                    formID={ONYXKEYS.FORMS.EDIT_PASSWORD_FORM}
+                    validate={this.validate}
+                    onSubmit={this.submit}
+                    submitButtonText={this.props.translate('common.save')}
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:94]: I think that we should change the text to `Change password`

File: src/pages/settings/EditPasswordScreen.js [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+                            secureTextEntry
+                            autoCompleteType="password"
+                            textContentType="password"
+                            returnKeyType="done"
+                            inputID="currentPassword"
+                        />
+                    </View>
+                    <View style={styles.mb6}>
+                        <TextInput
+                            label={`${this.props.translate('passwordPage.newPassword')}*`}
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:108]: This seems to be the only place that uses this translation key. Can we just move `*` to the translation files and simplify the code here?
```suggestion
                            label={this.props.translate('passwordPage.newPassword')}
```

File: src/pages/settings/EditPasswordScreen.js [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+                            autoCompleteType="password"
+                            textContentType="password"
+                            inputID="newPassword"
+                        />
+
+                        <Text
+                            style={[
+                                styles.textLabelSupporting,
+                                styles.mt1,
+                            ]}
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:119]: I'd make this one line

File: src/pages/settings/EditPasswordScreen.js [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+                        >
+                            {this.props.translate('passwordPage.newPasswordPrompt')}
+                        </Text>
+
+                    </View>
+                    { !_.isEmpty(this.props.account.errors) && (
+                        <Text style={styles.formError}>
+                            {ErrorUtils.getLatestErrorMessage(this.props.account)}
+                        </Text>
+                    )}
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:129]: We shouldn't need this. Form will automatically display the errors passed to the FormID key. I'm not sure that we need to subscribe to the account Onyx key.

File: src/pages/settings/EditPasswordScreen.js [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+
+export default compose(
+    withLocalize,
+    withOnyx({
+        account: {
+            key: ONYXKEYS.ACCOUNT,
+        },
+        requestCallForm: {
+            key: ONYXKEYS.FORMS.REQUEST_CALL_FORM,
+        },
```
Comment (luacmartins) [inline src/pages/settings/EditPasswordScreen.js:147]: Why are we subscribing to this key?

File: src/pages/settings/ResendValidationLinkScreen.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+import ResendValidationMessage from './ResendValidationMessage';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+};
+
+class ResendValidationLinkScreen extends Component {
```
Comment (luacmartins) [inline src/pages/settings/ResendValidationLinkScreen.js:17]:
```suggestion
class ResendValidationLinkPage extends Component {
```

File: src/pages/settings/ResendValidationMessage.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
+import Str from 'expensify-common/lib/str';
```
Comment (luacmartins) [inline src/pages/settings/ResendValidationMessage.js:1]: I don't think that we are reusing this component anywhere. Maybe we can just move this logic to `ResendValidationLinkPage`

File: src/pages/settings/UnauthenticatedUser.js [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+};
+
+class UnauthenticatedUser extends Component {
+    constructor(props) {
+        super(props);
+        this.onResendValidationLink = this.onResendValidationLink.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.account.isLoading || this.props.account.isLoading) { return; }
```
Comment (luacmartins) [inline src/pages/settings/UnauthenticatedUser.js:41]:
```suggestion
        if (!prevProps.account.isLoading || this.props.account.isLoading) {
            return;
        }
```

**Review by luacmartins — CHANGES_REQUESTED** [review]
Left a preliminary review.

File: src/pages/settings/EditPasswordScreen.js (hunk 6 above)
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:28]: @luacmartins  I think we should keep account prop to show the error here: https://github.com/Expensify/App/blob/533e63ce8b13e573024784836cef0bb7e859e937/src/pages/settings/PasswordPage.js#L211

File: src/pages/settings/EditPasswordScreen.js (hunk 11 above)
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:108]: It seems to be weird when we change newPassword: 'New password*'. That makes things confused

File: src/pages/settings/EditPasswordScreen.js (hunk 13 above)
Comment (tienifr) [reply src/pages/settings/EditPasswordScreen.js:129]: Screen Shot 2022-11-04 at 15 43 12

As I commented here https://github.com/Expensify/App/pull/12032#discussion_r1013727840, I think we still need to subscribe to the account to get the error.

File: src/pages/settings/ResendValidationMessage.js (hunk 16 above)
Comment (tienifr) [reply src/pages/settings/ResendValidationMessage.js:1]: I just use `Str` to check the user logins by email or sms: 
`const isSMSLogin = Str.isSMSLogin(props.credentials.login);`

**Comment by tienifr** [comment]
@luacmartins Thanks for your review. I've change my PR, but there are some points that we need to discuss more, pls let me know what you think?

File: src/pages/settings/EditPasswordScreen.js (hunk 6 above)
Comment (luacmartins) [reply src/pages/settings/EditPasswordScreen.js]: Makes sense. I forgot that the API is still returning the errors to the account key.

File: src/pages/settings/EditPasswordScreen.js (hunk 11 above)
Comment (luacmartins) [reply src/pages/settings/EditPasswordScreen.js]: Hmm actually I don't think that we need `*` at all. We don't use that in any required fields.

Comment (luacmartins) [reply src/pages/settings/EditPasswordScreen.js]: I think we should remove those from Password page as well

File: src/pages/settings/EditPasswordScreen.js (hunk 13 above)
Comment (luacmartins) [reply src/pages/settings/EditPasswordScreen.js]: Yea, that's ok. Ideally the API would return the error to the `EDIT_PASSWORD_FORM` key, but I'm not sure if we need to refactor that right now so we can keep it.

**Comment by luacmartins** [comment]
@tienifr addressed your comments

File: src/pages/settings/ResendValidationMessage.js (hunk 16 above)
Comment (luacmartins) [reply src/pages/settings/ResendValidationMessage.js:1]: I meant the `ResendValidationMessage.js` component. I'm not sure that we should make it its own separate component

**Comment by tienifr** [comment]
@luacmartins I've updated my PR, pls help check again. Thanks.

**Comment by tienifr** [comment]
@luacmartins I thought I resolved all your comments. Pls let me know if I miss something

**Comment by luacmartins** [comment]
@tienifr ah nice! sorry, I missed that. Just left a couple more comments, before I start testing.

**Comment by tienifr** [comment]
@luacmartins Oh my bad, I've updated my PR. Thanks for pointing that out

**Comment by luacmartins** [comment]
@tienifr I'll test once the merge conflict is resolved!

**Comment by luacmartins** [comment]
@tienifr please avoid force pushing commits. It makes it hard to follow the commit history and it can be confusing for reviewers.

**Comment by luacmartins** [comment]
As per [this comment](https://github.com/Expensify/App/issues/11427#issuecomment-1307978456), this PR is no longer needed. I'll close this and we will move forward with compensation on the respective issue. Thanks for the work @tienifr!
